ocaml/xapi/helpers.ml

```diff
@@ -42,21 +42,15 @@ let log_exn_continue msg f x =
 type log_output = Always | Never | On_failure
 
 let filter_patterns =
-  [
-    ( Re.Str.regexp "^\\(.*proxy_\\(username\\|password\\)=\\)\\(.*\\)$"
-    , "\\1(filtered)"
-    )
-  ]
+  [(Re.Pcre.regexp "^(.*proxy_(username|password)=)(.*)$", "(filtered)")]
```

Comment:
so with this change you are relying on the fact that `t` consists of `1` and `(filtered)`, right? What if the `t` is something more complicated like `xxx\1yy\2zz\4`, it seems that this would make the t less expressive?

Reply:
🆗